src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -482,44 +482,78 @@ public static bool TryValidateBlock(BinaryReader reader, ZipArchiveEntry entry)
             }
             else
             {
-                if (reader.BaseStream.Length < reader.BaseStream.Position + extraFieldLength + entry.CompressedLength + 4)
+                // There are 4 distinct scenario we would like to support
+                //   1.based on the appnote it seems that the structure of this record is following:
+                //                  crc-32                          4 bytes
+                //                  compressed size                 4 bytes (scenario 1.a has 8 bytes)
+                //                  uncompressed size               4 bytes (scenario 1.a has 8 bytes)
+                //
+                //   2.based on files that we have been able to examine
+                //                  data descriptor signature        4 bytes  (0x08074b50)
+                //                  crc-32                                  4 bytes
+                //                  compressed size                   4 bytes (scenario 2.a has 8 bytes)
+                //                  uncompressed size               4 bytes (scenario 2.a has 8 bytes)
+                //
+                // we can safely assume that this record is not the last one in the file, so let's just
```

Comment:
```suggestion
                // We can safely assume that this record is not the last one in the file, so let's just
```
It is nice to start a new sentence with a capital letter (I am not commenting on all places)